### PR TITLE
Remove type-unlimited unary `+` and `*`

### PR DESCRIPTION
**Pull request by dkarrasch**
Addresses https://github.com/JuliaLang/julia/pull/44564#issuecomment-1126987158.

### CODE REVIEW

**Comment by dkarrasch** [comment]
@nanosoldier `runtests(ALL, vs = ":master")`

**Comment by bkamins** [comment]
I am moving the discussion to this PR from #44564. CC @nalimilan 

> I'm very sorry for the breakage!

No problem - that is why I am reporting.

---

The point is that `-` is different than `*` and `+` because in `-` you multiply by `UInt8(-1)` which will error if something is undefined, while `*` and `+` silently go-through even when they do not make sense to be allowed.

I do not insist on this PR, but I would want to have a conscious decision that we want to allow these operations (in DataFrames.jl I strictly test if nonsense operations error - that is how I got to this PR 😄).

Specifically in Julia 1.7, since addition of `NamedTuple` in general is disallowed then also:
```
julia> +(NamedTuple())
ERROR: MethodError: no method matching +(::NamedTuple{(), Tuple{}})
```
while after your PR:
```
julia> +(NamedTuple())
NamedTuple()
```
The point is that, in general you call addition like `+(args...)` and in generic code you do not know how many `args` you get. Previously if all `args` were `NamedTuple` the code always errored. Now it errors for more than one arg, but for one arg it works.

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/45320#issuecomment-1127459651) has completed - possible new issues were detected. A full report can be found [here](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/f3c8244_vs_1ee1bbd/report.html).

File: base/operators.jl [hunk 1]
```diff
@@ -512,10 +512,10 @@ julia> identity("Well, what did you expect?")
 """
 identity(@nospecialize x) = x
 
-+(x) = x
++(x::Number) = x
 -(x) = Int8(-1)*x
```
Comment (StefanKarpinski) [inline base/operators.jl:516]: Kind of seems like all of these other type-unlimited arithmetic fallbacks should also be done away with.

**Pull request merged**